diff --git a/Cargo.toml b/Cargo.toml
index 154667a..e628371 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -34,10 +34,10 @@ dolos-minibf = { path = "crates/minibf", optional = true }
 gasket = { version = "^0.9", features = ["derive"] }
 # gasket = { path = "../../construkts/gasket-rs/gasket", features = ["derive"] }
 
-tx3-lang = "0.6.0"
-tx3-cardano = "0.6.0"
-# tx3-lang = { path = "../../tx3-lang/tx3/crates/tx3-lang" }
-# tx3-cardano = { path = "../../tx3-lang/tx3/crates/tx3-cardano" }
+# tx3-lang = "0.6.0"
+# tx3-cardano = "0.6.0"
+tx3-lang = { path = "../tx3/crates/tx3-lang" }
+tx3-cardano = { path = "../tx3/crates/tx3-cardano" }
 # tx3-lang = { git = "https://github.com/tx3-lang/tx3.git" }
 # tx3-cardano = { git = "https://github.com/tx3-lang/tx3.git" }
 
